### PR TITLE
Feature/dependency update

### PR DESCRIPTION
**Pull request by outsidecoder**
### What

Added CI folder, and updated Makefile and merged branch feature/library-pipelines to allow audit to pass on pipeline.

### How to review

Review changes, and run tests, audit and build make options in Makefie.

### Who can review

Anyone

### CODE REVIEW

**Review by CarlHembrough — APPROVED** [review]

File: ci/build.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.15
+    tag: latest
```
Comment (redhug1) [inline ci/build.yml:9]: We need to specify a specifc version of 'go' for traceability and tracking down bugs, please revert this.

File: ci/unit.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.15
+    tag: latest
```
Comment (redhug1) [inline ci/unit.yml:9]: We need to specify a specifc version of 'go' for traceability and tracking down bugs, please revert this.

**Review by CarlHembrough — CHANGES_REQUESTED** [review]
Please see comment about the version of Go

**Review by DavidSubiros — COMMENTED** [review]
Changes look good to me. Just a small comment: could you run `go mod tidy` please? (so that it removes unnecessary dependencies from go.sum)

**Review by CarlHembrough — APPROVED** [review]

**Comment by DavidSubiros** [comment]
I think that the vulenrable dependency comes from graphson. I am working on grahpson at the moment for another task , I will fix the dependency there as well.

File: ci/build.yml (hunk 1 above)
Comment (outsidecoder) [reply ci/build.yml:9]: As per discussion - this is a separate pipeline for libraries, so taking the latest version is preferred as an early warning to potential issues up ahead.

File: ci/unit.yml (hunk 2 above)
Comment (outsidecoder) [reply ci/unit.yml:9]: As per discussion - this is a separate pipeline for libraries, so taking the latest version is preferred as an early warning to potential issues up ahead.

**Comment by outsidecoder** [comment]
> I think that the vulenrable dependency comes from graphson. I am working on grahpson at the moment for another task , I will fix the dependency there as well.
> Update: Actually, I think it comes from go convey (used golang crypto version has a vulnerability) - how did we handle this in other repos that also use go-convey v1.6.4? (latest)

I did a `go get -u golang.org/x/crypto` and that fixed that dependency.

File: ci/unit.yml (hunk 2 above)
Comment (redhug1) [reply ci/unit.yml:9]: OK

**Review by redhug1 — APPROVED** [review]
OK

**Review by redhug1 — APPROVED** [review]
testing approve

**Pull request merged**